### ISSUE TITLE
Add settings object

### ISSUE THREAD

**Issue by berislavlopac** [berislavlopac/figga#4]
Right now, the settings are supposed to be read from a source and kept in (a kind of) a dictionary. The typing is limited, and there is no enforcing types if the source keeps everything in string, like it's the case for environment variables or `ini` files.

The idea is to add the support for specifying a set of predefined settings, in a form similar to a [`dataclass`](https://docs.python.org/3.8/library/dataclasses.html#dataclasses.dataclass) or [`attr.s`](https://www.attrs.org), something like:

```python
@dataclass
class Config:
    some_value: int
    some_value_with_default: str = "foo bar"

config = configuration(Config)
```
... or something similar.

Basically it would "enrich" the class with various functionalities useful for configuration:

* Methods to ingest values from files and environment, similar to what we have now, but with full conversion to the right types.
* Option to allow or ban values not defined in the specification, and how to convert them.
* Automatic construction of [argument parsers](https://docs.python.org/3.8/library/argparse.html) for CLI applications.
